tools/zypp-CheckAccessDeleted.cc

```diff
@@ -94,33 +94,59 @@ struct ProcInfoTable
+{
+  std::cout << "Usage: " << appname << " [--help]" << std::endl;
+  std::cout << "List information about all running processe" << std::endl;
+  std::cout << "which access deleted executables or libraries." << std::endl;
+  std::cout << "  PID     " << "process ID" << std::endl;
+  std::cout << "  PPID    " << "parent process ID" << std::endl;
+  std::cout << "  UID     " << "process user ID" << std::endl;
+  std::cout << "  LOGIN   " << "process login name" << std::endl;
+  std::cout << "  COMMAND " << "process command name" << std::endl;
+  std::cout << "  SERVICE " << "/etc/init.d/ script that might be used to restart the command (guessed)" << std::endl;
+  std::cout << "  FILES   " << "list of deleted executables or libraries accessed" << std::endl;
+}
+
 int main( int argc, char * argv[] )
 {
-  if ( argc >= 2 )
-  {
-    std::string progname( zypp::Pathname::basename( argv[0] ) );
-    if ( strcmp( argv[1], "--help" ) == 0 )
+
+  std::string progname( zypp::Pathname::basename( argv[0] ) );
+  std::string debugInputFile;
+  argv++;
+  argc--;
+
+  while( argc > 0 ) {
+    if ( strcmp( argv[0], "--help" ) == 0 )
     {
-      std::cout << "Usage: " << progname << " [--help]" << std::endl;
-      std::cout << "List information about all running processe" << std::endl;
-      std::cout << "which access deleted executables or libraries." << std::endl;
-      std::cout << "  PID     " << "process ID" << std::endl;
-      std::cout << "  PPID    " << "parent process ID" << std::endl;
-      std::cout << "  UID     " << "process user ID" << std::endl;
-      std::cout << "  LOGIN   " << "process login name" << std::endl;
-      std::cout << "  COMMAND " << "process command name" << std::endl;
-      std::cout << "  SERVICE " << "/etc/init.d/ script that might be used to restart the command (guessed)" << std::endl;
-      std::cout << "  FILES   " << "list of deleted executables or libraries accessed" << std::endl;
+      usage( progname );
       return 0;
+    } else if ( strcmp( argv[0], "--fromDebugFile" ) ) {
+      if ( argc < 2 ) {
+        std::cerr << progname << ": fromDebugFile requires a argument" << std::endl;
+        return 1;
+      }
+
+      argv++;
+      argc--;
+      debugInputFile = argv[0];
+
+    } else {
+      std::cerr << progname << ": unexpected argument '" << argv[0] << "'" << std::endl;
+      std::cerr << "Try `" << progname << " --help' for more information." << std::endl;
+      return 1;
     }
-    std::cerr << progname << ": unexpected argument '" << argv[1] << "'" << std::endl;
-    std::cerr << "Try `" << progname << " --help' for more information." << std::endl;
-    return 1;
+
+    argv++;
+    argc--;
   }
 
   zypp::CheckAccessDeleted checker(false); // wait for explicit call to check()
   try {
-    checker.check( /*verbose*/true );
+    if ( debugInputFile.empty() )
+      checker.check( /*verbose*/true );
+    else
+      checker.check( debugInputFile, /*verbose*/true );
```

Comment:
Just realized and remembered, that `CheckAccessDeleted` has a _normal_ (zypper ps) mode, and a _verbose_ one.

Maybe spend a `--zypper` option to be able to filter the same way as zypper does (verbose=false). Should be mentioned in `usage` and also in the man page (`doc/zypp-CheckAccessDeleted.1.txt`). 

*man pages:* See [trello](https://trello.com/c/XrPhiLqP/148-fix-man-page-building-in-libzypp-and-zypper). We currently ship the manpage in the git repo (creating it locally before check-in). If we fix this first, it's sufficient to update the asciidoc file as we'll no longer need to ship the created manpage too.